src/main/java/com/bugsnag/servlet/BugsnagServletRequestListener.java

```diff
@@ -25,4 +30,12 @@ public void requestInitialized(ServletRequestEvent servletRequestEvent) {
     public void requestDestroyed(ServletRequestEvent servletRequestEvent) {
         SERVLET_REQUEST.remove();
     }
+
+    private void trackServletSession() {
+        for (Bugsnag bugsnag : Bugsnag.uncaughtExceptionClients()) {
```

Comment:
If multiple bugsnag clients have been initialised, each of these will be informed that a HTTP request has been made, and record a session each if necessary. There should only ever be 1 session recorded for each client though.

Did you have specific concerns about performance here? This is definitely a hot path so it might be worth profiling, but at an initial glance I can't see too many expensive operations apart from possibly UUID generation/calendar usage.

I think this does mean that it can only track sessions when exceptions are being automatically caught.

Reply:
This is definitely the behaviour we want - I've forwarded you an email with relevant design discussion